dom.go

```diff
@@ -132,6 +134,8 @@ func (h *HTML) restoreHTML(prev *HTML) {
 		}
 	}
 
+	h.populateClassNameProperty()
```

Comment:
In this situation, I believe the behavior isn't correct. This would cause us to set `className` on the node every single time anything on the element is changed -- regardless of whether or not `className` itself has actually changed since the last render. That violates the standard reconciliation pattern being used here.

What is the motivation for moving `className` property setting into `*HTML` as if it's a first-class thing? After all, `className` is just a standard property like everything else -- there is no need for us to do this here AFAIK.